docs/src/languages/php/_index.md

```diff
@@ -22,261 +22,154 @@ Note that from PHP 7.1, the images use the Zend Thread Safe (ZTS) version of PHP
 
-* Have your script `include` dependencies instead of `require`
-  and fail gracefully if the dependencies aren't there.
-* Enable preloading with a variable that [isn't available during the build](../../development/variables/set-variables.md#variable-options).
-  Then preloading happens only on deploy.
+You can remove the default build flavor and run your own commands for complete control over your build.
+Set the build flavor to `none` and add the commands you need to your `build` hook, as in the following example:
 
-## FFI
+```yaml {location=".platform.app.yaml"}
+build:
+    flavor: none
 
-PHP 7.4 introduced support for Foreign Function Interfaces (FFI),
-which allows user-space code to bridge to existing C-ABI-compatible libraries.
-FFI is fully supported on Platform.sh.
+hooks:
+    build: |
+        set -e
+        composer install --no-interaction --no-dev
+```
 
-Note: FFI is only intended for advanced use cases, and is rarely a net win for routine web requests.
-Use with caution.
+That installs production dependencies with Composer but not development dependencies.
+The same can be achieved by using the default build flavor and [adding the `COMPOSER_NO_DEV` variable](../../development/variables/set-variables.md#create-environment-specific-variables).
 
-There are a few steps to leveraging FFI:
+See more on [build flavors](../../create-apps/app-reference.md#build).
 
-1. Enable the FFI extension in `.platform.app.yaml`:
+### Alternative repositories
 
-    ```yaml
-    runtime:
-        extensions:
-            - ffi
-   ```
+In addition to the standard `dependencies` format,
+you can specify alternative repositories for Composer to use as global dependencies.
+So you can install a forked version of a global dependency from a custom repository.
 
-2. Specify a [preload file](#opcache-preloading) in which you can call `FFI::load()`.
-   Using `FFI::load()` in preload is considerably faster than loading the linked library on each request or script run.
+To install from an alternative repository:
 
-3. Ensure the library is available locally, but not in a web-accessible directory.
-   `.so` files may included in your repository, downloaded in your build hook, or compiled in your build hook.
-   If compiling C code, `gcc` is available by default.
-   If compiling Rust code, you can download the [Rust compiler in the build hook](https://doc.rust-lang.org/stable/book/ch01-01-installation.html).
+1. Set an explicit `require` block:
 
-4. For running FFI from the command line,
-   you need to enable the OPcache for command line scripts in addition to the preloader.
-   The standard pattern for the command would be `php -d opcache.preload="your-preload-script.php" -d opcache.enable_cli=true your-cli-script.php`.
+    ```yaml {location=".platform.app.yaml"}
+    dependencies:
+        php:
+            require:
+                "platformsh/client": "2.x-dev"
+    ```
```

Comment:
Hmmm, I don't think I get that part nor the explanation 🤔 
I usually go with a tab, my tab is set to 4 spaces.

Also we might want to disable the maximum line length rule because of the shortcodes: "remark-lint-maximum-line-length".